### DISCUSSION TITLE
How to change Address Space order

### DISCUSSION THREAD

**Discussion by MatthewBoss96** [FreeOpcUa/opcua-asyncio#1192] [Q&A] [answered]
How can I change the order in which Objects/Methods/Variables are displayed in the Address Space? Currently they are ordered in an alphabetical order, but I want to have control over the order in which the elements are shown in the Address Space.
I checked whether it could be solved by using their identifiers, but this did not have any effect on the order thus far.
Could anyone please help me out?

**Comment by AndreasHeine** [comment, marked as answer]
its opcua client dependend how they order the browse results... the opc ua server does not care about such things!

the addressspace is a unsorted key:value store on serverside

**Reply by MatthewBoss96** [reply]
Hi Andreas, I already thought something like that, but I cannot find anything related to the address space when using Settings -> Configure UA Expert. I also checked the Server and the Address_space classes, but I do not have a clue how I can use these to set the order to something different than alphabetical?

**Reply by AndreasHeine** [reply]
Ua-Expert uses the "BrowseNames" in alphabetical order to display the tree there is nothing i know you can change about it -> UaExpert sends a BrowseRequest (e.g. Objects-Node) and gets a BrowseResponse back which is simplified a List of Nodes/NodeIds and UaExpert just sorts them according to the BrowseName... not sure why its important for you? because a ua system usually don't care about at all!

**Reply by MatthewBoss96** [reply]
Okay thanks for the information. An alphabetical order is less intuitive in my current situation where I would desire to have more control over the order of my objects, but its not a big deal. I was just wondering if this option was available or if it could become a feature request.
